packit_service/worker/events/event.py

```diff
@@ -465,7 +465,47 @@ def get_package_config(self) -> Optional[PackageConfig]:
         return package_config
 
     @staticmethod
-    def _filter_failed_models_targets(
+    def _get_submitted_time_from_model(
+        model: Union[CoprBuildTargetModel, TFTTestRunTargetModel]
+    ) -> datetime:
+        # TODO: unify `submitted_name` (or better -> create for both models `task_accepted_time`)
+        # to delete this mess plz
+        if isinstance(model, CoprBuildTargetModel):
+            return model.build_submitted_time
+
+        return model.submitted_time
+
+    @classmethod
+    def get_most_recent_targets(
+        cls,
+        models: Union[
+            Optional[Iterable[CoprBuildTargetModel]],
+            Optional[Iterable[TFTTestRunTargetModel]],
+        ],
+    ) -> Dict[str, datetime]:
+        """
+        Gets most recent targets (regarding submission time).
+
+        Args:
+            models: Copr ot TF models
+
+        Returns:
+            Dictionary - target as a key and corresponding most recent submission time as a value.
+        """
+        target_submission_time: Dict[str, datetime] = {}
+        for model in models:
+            submitted_time = cls._get_submitted_time_from_model(model)
+            if (
```

Comment:
it took me a while to get what this function really does :D would it work to have Dict[str, "TargetModel"] and then return directly the values of the dictionary and in the `_filter..` methods no filtering on time would be needed anymore? 
(in the check we would compare then `cls._get_submitted_time_from_model(the_dictionary.get(model.target))` with the current submitted time) Then the method name could stay how it is now